.github/workflows/test.yml

```diff
@@ -51,12 +52,13 @@ jobs:
           "ubuntu-py39",
           "ubuntu-py310",
           "ubuntu-py311",
+          "ubuntu-py312",
           "ubuntu-pypy3",
 
           "macos-py37",
-          "macos-py38",
```

Comment:
Probably good to leave a comment here explaining why we are skipping macos-py38.

Reply:
I noticed macos-py311 was missing, so I added py312 and removed py38 to keep the job count the same. With the faster Python release cycle, it is starting to become a lot.